lsv_compassion/models/payment_order.py

```diff
@@ -111,43 +108,41 @@ def _prepare_bank_payment_line(self, paylines):
 
     @api.model
     def _prepare_bank_payment_line_with_lang(self, paylines, partner, res):
-        with setlocale(partner.lang):
-            invoices = paylines.mapped('move_line_id.invoice_id').with_context(
-                lang=partner.lang)
-            products = invoices.mapped('invoice_line_ids.product_id')
-            invoice_type = list(set(invoices.mapped('invoice_type')))
-            invoice_type = invoice_type and invoice_type[0] or 'other'
-            communication = False
-            if invoice_type in ('sponsorship', 'gift'):
-                children = invoices.mapped(
-                    'invoice_line_ids.contract_id.child_id')
-                if len(children) == 1:
-                    if invoice_type == 'sponsorship':
-                        communication = _('Sponsorship')
-                    elif len(products) == 1:
-                        communication = products.name + ' ' + _('for')
-                    else:
-                        communication = _('sponsorship gifts').title() + \
-                            ' ' + _('for')
-                    communication += ' ' + children.preferred_name
+        invoices = paylines.mapped('move_line_id.invoice_id').with_context(
```

Comment:
Here you should create a local dict context containing the language, otherwise the translation function `_()` won't translate in the right language at line 122.
```suggestion
        context = {'lang': partner.lang}
        invoices = paylines.mapped('move_line_id.invoice_id').with_context(context)
```